src/quemb/shared/helper.py

```diff
@@ -66,3 +71,34 @@ def ncore_(z: int) -> int:
         print("exiting", flush=True)
         sys.exit()
     return nc
+
+
+def write_cube(
+    be_object: object, cube_file_path: PathLike, fragment_idx: Optional[list], **kwargs
+) -> None:
+    """Write cube files of embedding orbitals from a BE object.
+
+    Parameters
+    ----------
+    be_object : object
+        BE object containing the fragments, each of which contains embedding orbitals.
+    cube_file_path : PathLike
+        Directory to write the cube files to.
+    fragment_idx : Optional[list]
+        Index of the fragments to write the cube files for. If None, all fragments are written.
+    kwargs: Any
+        Keyword arguments passed to cubegen.orbital.
+    """
+    assert type(be_object) is molbe.BE, NotImplementedError(
```

Comment:
# isinstance
I would use `isinstance`, that also works for subclasses.

# assert
In addition, there is one more important thing about `assert` that I did not have time to mention.
`assert` can be stripped off when invoking optimized builds with the `-O` option.
This means:
- It is great to use `assert` for things that we assume to be true in the code. One can even put semi-compute-intensive tests into `assert` to have a slower debug build and a fast optimized release. (The difference is of course more striking for compiled languages.
- It is dangerous to use assert for user validation, if you want it to always be executed. 
- assert will raise an `AssertionError` and accepts a string as second argument. I would not change this behavoiur.

All in all, I would rewrite as:

```python3
if not isinstance(be_object, molbe.BE):
    raise NotImplementedError(...)
```

Reply:
addressed this -- thanks!